orchagent/vnetorch.cpp

```diff
@@ -249,6 +259,550 @@ VNetVrfObject::~VNetVrfObject()
     SWSS_LOG_INFO("VNET '%s' deleted ", vnet_name_.c_str());
 }
 
+/*
+ * Bitmap based VNET class definition
+ */
+uint32_t VNetBitmapObject::vnetBitmap_ = 0;
+set<uint32_t> VNetBitmapObject::tunnelOffsets_;
+map<string, uint32_t> VNetBitmapObject::vnetIds_;
+map<uint32_t, VnetBridgeInfo> VNetBitmapObject::bridgeInfoMap_;
+map<tuple<MacAddress, sai_object_id_t>, sai_fdb_entry_t> VNetBitmapObject::fdbMap_;
+map<tuple<MacAddress, sai_object_id_t>, sai_neighbor_entry_t> VNetBitmapObject::neighMap_;
+
+VNetBitmapObject::VNetBitmapObject(const std::string& vnet, const VNetInfo& vnetInfo,
+                             vector<sai_attribute_t>& attrs) : VNetObject(vnetInfo)
+{
+    SWSS_LOG_ENTER();
+
+    setVni(vnetInfo.vni);
+
+    vnet_id_ = getFreeBitmapId(vnet);
+}
+
+bool VNetBitmapObject::updateObj(vector<sai_attribute_t>&)
+{
+    SWSS_LOG_ENTER();
+
+    return false;
+}
+
+uint32_t VNetBitmapObject::getFreeBitmapId(const string& vnet)
+{
+    SWSS_LOG_ENTER();
+
+    for (uint32_t i = 0; i < VNET_BITMAP_SIZE; i++)
+    {
+        uint32_t id = 1 << i;
+        if ((id & vnetBitmap_) == 0)
+        {
+            vnetBitmap_ |= id;
+            vnetIds_.emplace(vnet, id);
+            return id;
+        }
+    }
+
+    return 0;
+}
+
+uint32_t VNetBitmapObject::getBitmapId(const string& vnet)
+{
+    SWSS_LOG_ENTER();
+
+    if (vnetIds_.find(vnet) == vnetIds_.end())
+    {
+        return 0;
+    }
+
+    return vnetIds_[vnet];
+}
+
+void VNetBitmapObject::recycleBitmapId(uint32_t id)
+{
+    SWSS_LOG_ENTER();
+
+    vnetBitmap_ &= ~id;
+}
+
+uint32_t VNetBitmapObject::getFreeTunnelRouteTableOffset()
+{
+    SWSS_LOG_ENTER();
+
+    for (uint32_t i = 0; i < 256; i++)
+    {
+        if (tunnelOffsets_.count(i) == 0)
```

Comment:
My preference would be to use a simple Boolean array of 256/512 for this logic. Just a suggestion.

Reply:
Done